mkosi/__init__.py

```diff
@@ -6273,6 +6273,37 @@ def run_ssh(args: CommandLineArguments) -> None:
         )
 
 
+def generate_secure_boot_key(args: CommandLineArguments) -> NoReturn:
+    """Generate secure boot keys using openssl"""
+    for f in (args.secure_boot_key, args.secure_boot_certificate):
+        if f is not None and not os.path.exists(f):
+            die(
+                f"{f} already exists.\n"
+                "If you are sure you want to generate new secure boot keys\n"
+                "remove {args.secure_boot_key} and {args.secure_boot_certificate} first."
+            )
+
+    os.execvp(
+        "openssl",
+        [
+            "openssl",
+            "req",
+            "-new",
+            "-x509",
+            "-newkey",
+            "rsa:2048",
+            "-keyout",
+            args.secure_boot_key or "./mkosi.secure-boot.key",
+            "-out",
+            args.secure_boot_certificate or "./mkosi.secure-boot.crt",
+            "-days",
+            "3650",
+            "-subj",
+            "/CN=mkosi/",
```

Comment:
I'd probably include the local user name. i.e. maybe generate `mkosi of lennart` for a user lennart  or something like that.

I think the validity and the CN are something that might be worth making configurable in new `mkosi.default` keys.

Reply:
Added a note about the CN, which defaults to `mkosi of user`, where user goes through `SUDO_USER`, `USER` and defaults to `unknown user`.

I've so far decided not to add any configurability.